File: cpu_usage_indicator/c++/cpu_usage_indicator.cxx
#include <unistd.h>

#include <csignal>
#include <iostream>

#include "proc_stat.hxx"
#include "led.hxx"


using namespace std;


class App {
	public:
		App();
		void run();
	private:
		unsigned int usage;
		ProcStat stat;
		Led ledr;
		Led ledg;
};


App::App(void)
	:stat(),
	 ledr("/sys/class/leds/sunxi_led0r/brightness" , 0),
	 ledg("/sys/class/leds/sunxi_led0g/brightness" , 0)
{

}

void App::run(void)
{
	usleep(800000);

	ledr.set_brightness(0);
	ledg.set_brightness(0);

	usleep(200000);

	this->usage = stat.get_cpu_usage();
	cout << "\rcpu usage = " << this->usage << " %  " << flush;

	if(this->usage < 30) {
		ledg.set_brightness(30);
	} else if(this->usage < 60) {
		ledg.set_brightness(5);
	} else if(this->usage < 90) {
		ledr.set_brightness(15);
	} else {
		ledr.set_brightness(150);
	}
}


App app;

volatile int running = 1;

void stop(int signo)
{
	running = 0;
}

int main()
{
	signal(SIGINT, stop);

	while(running) {
		app.run();
	}
	cout << endl;
}

File: cpu_usage_indicator/nodejs/cpu_usage_indicator.js
var Led = require('./led');
var ProcStat = require('./proc_stat');

var ledr = new Led('/sys/class/leds/sunxi_led0r/brightness');
var ledg = new Led('/sys/class/leds/sunxi_led0g/brightness');
var stat = new ProcStat();

ledr.setBrightness(0);
ledg.setBrightness(0);

var ms = 0;

var timer = setInterval(function () {
	ms += 200;

	if(ms == 800) {
		ledr.setBrightness(0);
		ledg.setBrightness(0);
	} else if(ms == 1000) {
		ms = 0;
		var usage = stat.getCpuUsage();
		process.stdout.write("\rcpu usage = " + usage + " %  ");

		if(usage < 30) {
			ledg.setBrightness(30);
		} else if(usage < 60) {
			ledg.setBrightness(5);
		} else if(usage < 90) {
			ledr.setBrightness(15);
		} else {
			ledr.setBrightness(150);
		}
	}
	

}, 200);

process.on('SIGINT', function () {
	clearInterval(timer);
	ledr.setBrightness(0);
	ledg.setBrightness(0);
	process.stdout.write("\n");
	process.exit();
});
